File: .tmp_c4000/batch_04_go/3487_maximum_unique_subarray_sum_after_deletion.go
// LeetCode 3487 - Maximum Unique Subarray Sum After Deletion
// https://leetcode.com/problems/maximum-unique-subarray-sum-after-deletion/

func maxSum(nums []int) int {
	seen := map[int]bool{}
	sum := 0
	hasPos := false
	maxNeg := int(-1e9)
	for _, x := range nums {
		if x < 0 {
			if x > maxNeg {
				maxNeg = x
			}
			continue
		}
		hasPos = true
		if !seen[x] {
			seen[x] = true
			sum += x
		}
	}
	if hasPos {
		return sum
	}
	return maxNeg
}

File: .tmp_rs4013/rust_check17/3487_maximum_unique_subarray_sum_after_deletion.rs
struct Solution;
// LeetCode 3487 - Maximum Unique Subarray Sum After Deletion
// https://leetcode.com/problems/maximum-unique-subarray-sum-after-deletion/

use std::collections::HashSet;

impl Solution {
    pub fn max_sum(nums: Vec<i32>) -> i32 {
        let mut seen = HashSet::new();
        let mut sum = 0;
        let mut has_pos = false;
        let mut max_neg = -1_000_000_000;
        for x in nums {
            if x < 0 {
                if x > max_neg {
                    max_neg = x;
                }
                continue;
            }
            has_pos = true;
            if seen.insert(x) {
                sum += x;
            }
        }
        if has_pos {
            sum
        } else {
            max_neg
        }
    }
}

fn main() {}
